espnet2/asr/preencoder/sinc.py

```diff
@@ -0,0 +1,245 @@
+#!/usr/bin/env python3
+#  2020, Technische Universität München;  Ludwig Kürzinger
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
+
+"""Sinc convolutions for raw audio input."""
+
+from collections import OrderedDict
+from espnet2.asr.preencoder.abs_preencoder import AbsPreEncoder
+from espnet2.layers.sinc_conv import LogCompression
+from espnet2.layers.sinc_conv import SincConv
+import humanfriendly
+import torch
+from typeguard import check_argument_types
+from typing import Optional
+from typing import Tuple
+from typing import Union
+
+
+class LightweightSincConvs(AbsPreEncoder):
+    """Lightweight Sinc Convolutions.
+
+    Provide a frontend for raw audio input.
+    https://arxiv.org/abs/2010.07597
+    """
+
+    def __init__(
+        self,
+        fs: Union[int, str, float] = 16000,
+        in_channels: int = 1,
+        out_channels: int = 256,
+        activation_type: str = "leakyrelu",
+        dropout_type: str = "dropout",
+    ):
+        """Initialize the module.
+
+        Args:
+            fs: Sample rate.
+            in_channels: Number of input channels.
+            out_channels: Number of output channels (for each input channel).
+            activation_type: Choice of activation function.
+            dropout_type: Choice of dropout function.
+        """
+        assert check_argument_types()
+        super().__init__()
+        if isinstance(fs, str):
+            fs = humanfriendly.parse_size(fs)
+        self.fs = fs
+        self.in_channels = in_channels
+        self.out_channels = out_channels
+        self.activation_type = activation_type
+        self.dropout_type = dropout_type
+
+        self.choices_dropout = {
+            "none": torch.nn.Identity,
+            "dropout": torch.nn.Dropout,
+            "spatial": SpatialDropout,
+            "dropout2d": torch.nn.Dropout2d,
+        }
+
+        self.choices_activation = {
+            "leakyrelu": torch.nn.LeakyReLU,
+            "relu": torch.nn.ReLU,
+        }
+
+        # initialization
+        self._create_sinc_convs()
+        self.init_sinc_convs()
+        # Sinc filters require custom initialization
+        self.espnet_initialization_fn = self.init_sinc_convs
```

Comment:
This is a bad way. It's enexpected and there are no reason to set method dynamically.

Please change `espnet_initialization_fn` to a defined method and remove init_sinc_convs.

```
def espnet_initialization_fn
```